### ISSUE TITLE
Support for `pnpm`

### ISSUE THREAD

**Issue by Teddarific** [Mirrorful/mirrorful#184]
Discovered by @karanvirsb!

Example error stack trace from `pnpm`:

```
Verbose mode enabled.

Running Mirrorful version: 4.0.1
Starting Mirrorful in D:\Documents\Shared\Coding\Test\mirrorful-test.

Current directory: D:\Documents\Shared\Coding\Test\mirrorful-test
New working directory: D:\Documents\Shared\Coding\Test\mirrorful-test\node_modules\mirrorful\editor
NextJS app detected.
Visit: http://localhost:5050/
Inside your project, you can run:

  yarn run mirrorful

to start the visual editor at any time 🚀

$ ../node_modules/.bin/next start -p 5050
node:internal/modules/cjs/loader:998
  throw err;
  ^

Error: Cannot find module 'D:\Documents\Shared\Coding\Test\next@12.3.4_biqbaboplfbrettd7655fr4n2y\node_modules\next\dist\bin\next'
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.12.1
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Done in 0.44s.
```

**Comment by ktra99** [comment]
I want to add to this issue a little bit from my own personal experience. I currently use `pnpm` for all my projects. One of the benefits of using `pnpm` is that instead of having to install all the packages over and over again, `pnpm` creates a store on your machine that stores the packages you install. If you later are using the same packages in a different project then `pnpm` simply refers to the already installed package in the store. Reducing installation time significantly. However this brings up concerns, especially since mirrorful is currently exporting based on a relative path - if I use `pnpm` to install mirrorful it will work as normal, however If I want to export tokens then those tokens will not be found in my project directory but rather placed relatively to the `pnpm` store if that makes sense

Here are some attempts of my own

- ❌  `npm install mirrorful -D` (Trying to install mirrorful with npm in a pnpm project results in) - `Cannot read properties of null (reading 'matches')`
- ❌  `pnpm add mirrorful -D` (Works but I expect to dive into my `pnpm` store somewhere on my machine, to find my beloved tokens)
- ✅  A workaround is to create a new directory inside your project and install mirrorful in isolation. Here's my [project](https://github.com/ktra99/digital-garden) for reference. I would ideally like to have everything within the same build. So looking forward to having `pnpm` supported!

**Comment by alexdanilowicz** [comment]
@ktra99 — very interesting. I know `pnpm` is getting very popular, so this is pretty important. thank you so much for sharing.

**Comment by alexdanilowicz** [comment]
@ktra99 is that what this commit is doing: https://github.com/ktra99/digital-garden/commit/a26668df945100fa02504939264ddb2b67404fc0 ? (Setting up mirrorful in isolation to use pnpm?)

**Comment by ktra99** [comment]
Yes exactly. I used npm to install mirrorful In a separate folder which worked and I'm now able to reference relatively closer to my application

**Comment by wouterfovel** [comment]
I'm using pnpm workspaces as well.

I was able to install and run mirrorful inside my pnpm workspaces setup, but it did not write files in the `.mirrorful` folder. Although it remembered its config. So, I assume mirrorful was writing its config and exported files some place else (looked for them, but did not found them :) ).

So, I'm using npm for mirrorful in my workspaces setup as a work-around and disabled pnpm from building it:

pnpm-workspaces.yaml
```
packages:
  # all packages in direct subdirs of modules/
  - 'modules/*'
  # all packages in subdirs of apps/
  - 'apps/**'
  # all packages in subdirs of tools/
  - 'tools/**'
  # exclude packages that are inside test directories
  - '!**/test/**'
  # TODO exclude design-tokens for now, it does not support pnpm. We should monitor the mirrorful project for this.
  - '!modules/design-tokens'
```

But would be great to have it build and operational with pnpm though :)

By the way, it's a great tool. Good job!!

**Comment by alexdanilowicz** [comment]
Hey thanks @wouterfovel! Really appreciate the message. Would you consider joining our Slack community: https://join.slack.com/t/mirrorful/shared_invite/zt-1ps2xtxh0-2NaixFfFzSKZbr5gw_AHfA? Would love to hear more about what you like about Mirrorful and how you're using it. Appreciate the kind words — we'd love to improve it for you!

**Comment by alexdanilowicz** [comment]
### update April 2023: 

It seems #358 got pnpm working for windows (thanks to our rockstar contributor @karanvirsb). Still working on Mac. Latest issue on Mac:

```bash
Error: Cannot find module '/Users/zeus/code/pnpm-next-m-13/node_modules/.pnpm/mirrorful@6.1.13_@babel+core@7.21.4_@chakra-ui+system@2.5.6_@types+node@18.16.1_@types+react@_xb3xch2zp4fu6lvbmuodpwzzj4/node_modules/.pnpm/next@13.2.4_@babel+core@7.21.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/bin/next'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:956:15)
    at Function.Module._load (node:internal/modules/cjs/loader:804:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
error Command failed with exit code 1.
➜  pnpm-next-m-13 git:(main) ✗ cd node_modules/.pnpm/mirrorful@6.1.13_@babel+core@7.21.4_@chakra-ui+system@2.5.6_@types+node@18.16.1_@types+react@_xb3xch2zp4fu6lvbmuodpwzzj4
➜  mirrorful@6.1.13_@babel+core@7.21.4_@chakra-ui+system@2.5.6_@types+node@18.16.1_@types+react@_xb3xch2zp4fu6lvbmuodpwzzj4 git:(main) ✗ cd node_modules/
➜  node_modules git:(main) ✗ ls
@chakra-ui          @vitejs             conf                fs                  open                posthog-js          update-check        vite                zustand
@emotion            chalk               cross-spawn         mirrorful           ora                 react               url                 vite-plugin-dts
@mirrorful          commander           framer-motion       next                path                react-dom           uuid                vite-tsconfig-paths
➜  node_modules git:(main) ✗ cd ./
./   ../
```

**Comment by alexdanilowicz** [comment]
@ktra99 and @wouterfovel are either of you on windows? Mind trying it out if so. `pnpm add -D mirrorful@latest` (6.1.13)

**Comment by karanvirsb** [comment]
Hey @alexdanilowicz I just made some changes that I think will fix the Mac and Linux version.

**Comment by alexdanilowicz** [comment]
This was fixed in the 16.1.15 release. Great job! 

FYI @ktra99 @wouterfovel 

Thanks @karanvirsb! 🎊

**Comment by wouterfovel** [comment]
Thanks a lot, it works with pnpm just great now :)
